mobile/src/people/departments/departments-horizontal-scrollable-list.tsx

```diff
@@ -2,40 +2,77 @@ import React, { Component, SyntheticEvent } from 'react';
 import { Animated, Easing, View, Text, ScrollView, Dimensions, TouchableOpacity, NativeScrollEvent, NativeSyntheticEvent } from 'react-native';
 import { AppState } from '../../reducers/app.reducer';
 import { EmployeeCardWithAvatar } from '../employee-card-with-avatar';
-import { DepartmentsTreeNode } from './departments-tree-node';
+import { DepartmentsTreeNode, stubIdForSubordinates } from './departments-tree-node';
 import { Employee } from '../../reducers/organization/employee.model';
 import { PeopleActions, updateDepartmentIdsTree } from '../../reducers/people/people.action';
 import { loadEmployeesForDepartment } from '../../reducers/organization/organization.action';
+import { EmployeesStore } from '../../reducers/organization/employees.reducer';
 
 interface DepartmentsHScrollableListProps {
     treeLevel?: number;
     departmentsTreeNodes?: DepartmentsTreeNode[];
     headDepartment: DepartmentsTreeNode;
+    employees?: EmployeesStore;
     topOffset?: number;
     requestEmployeesForDepartment: (departmentId: string) => void;
     updateDepartmentIdsTree: (index: number, department: DepartmentsTreeNode) => void;
     onItemClicked: (e: Employee) => void;
+    employeesPredicate: (employee: Employee) => boolean;
 }
 
 export class DepartmentsHScrollableList extends Component<DepartmentsHScrollableListProps> {
     private employeeCards: EmployeeCardWithAvatar[];
     private animatedValue: Animated.Value;
     private buttonText: Text;
-    private currentPage: number;
+    private currentPage: number = null;
+    private forceComponentRender = false;
 
     public componentDidMount() {
-        // this.animate.bind(this, Easing.bounce);
         if (this.props.departmentsTreeNodes !== null) {
             this.props.requestEmployeesForDepartment(this.props.departmentsTreeNodes[0].departmentId);
         }
     }
 
+    public shouldComponentUpdate(nextProps: DepartmentsHScrollableListProps) {
+        if (this.forceComponentRender) {
```

Comment:
it should be a pure function, why you need to change state?

Reply:
ok, i see why

Reply:
you should have currentPage in State instead, when you don't need to that. this function takes state as a second argument, I believe

Reply:
correct, I'll address it tomorrow.